src/babashka/cli.cljc

```diff
@@ -321,6 +325,19 @@
                     (-> {:spec spec :type :org.babashka/cli}
                         (merge data)
                         error-fn*))
+         [opt shell cmdline] args
+         _ (case opt
+             "--babashka.cli/completion-snippet"
+             (if-let [command-name (get-in opts [:completion :command])]
+               (do (print-completion-shell-snippet (keyword shell) command-name)
+                   (System/exit 0))
```

Comment:
We need to short circuit all following execution in this case though, since we're printing the shell completion snippet and any other code running after this might print stuff that messes this up. It only happens when you pass the internal `--babashka.cli/completion-snippet` opt, so regular user code should never hit this.

Reply:
You don't need to short circuit if you just put the logic after all the parsing logic though?
Also removing the System/exit stuff makes it more REPL friendly

Reply:
If I put the code after the parsing logic the consuming code could still print extra stuff. e.g.
```clojure
(defn -main [& args]
  (cli/parse-opts args opts)
  (println "this does not belong in the completions script"))
```

Reply:
The alternative would be to make it the responsibility of the user to call `print-completion-shell-snippet` (and make sure nothing else is printed in that case)

Reply:
> The alternative would be to make it the responsibility of the user to call 

Yeah, maybe that would be the least magic option, I was going to suggest this, but you already thought of it :)

Reply:
It's a bit brittle though imo because then it would also be the responsibility of the user to call `print-opts-completions` or `print-dispatch-completions` when their program is invoked with the `--org.babashka.cli/complete` option, making sure to pass the value of `:org.babashka.cli/complete` (i.e. the type of shell to complete for) and the value of the remaining arg (i.e. `(first (:args (cli/parse-opts …))` )

Reply:
We can make a default spec for that I guess? And the arg could become another option as well in the default spec?

Reply:
Yes, we could do that. But then the options show up in `format-opts`

Reply:
Sure, but then you'll have a `format-opts`, that magically swallows some options.
I also don't particularly like the magic in `parse-opts`, but at least that means you get completions without any setup on your part.
Changing `format-opts` means you still have magic and need to do setup.

Reply:
I'm also ok with just providing the `complete` * functions and having the user wire everything up themselves. Then it's even more work, but absolutely no magic.

Reply:
> Well, we control `format-opts` so we can decide to filter those options out, right?

Or do you mean the caller filters them out?

Reply:
> magically swallows some options

The fact that `org.babashka.cli/...` options are internal, makes this non-magic imo. Users of this library should never make up options with that namespace themselves